javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java

```diff
@@ -76,7 +77,30 @@ protected static void generateProtocolMethodJavadoc(ClientMethod clientMethod, J
             // Response body
             IType responseBodyType;
             if (JavaSettings.getInstance().isDataPlaneClient()) {
-                responseBodyType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                // special handling for paging method
+                if (clientMethod.getType() == ClientMethodType.PagingSync || clientMethod.getType() == ClientMethodType.PagingAsync || clientMethod.getType() == ClientMethodType.PagingAsyncSinglePage || clientMethod.getType() == ClientMethodType.PagingSyncSinglePage) {
+                    String itemName = clientMethod.getMethodPageDetails().getItemName();
+                    // rawResponseType has properties: 'value' and 'nextLink'
+                    IType rawResponseType = clientMethod.getProxyMethod().getRawResponseBodyType();
+                    ClientModel model = ClientModelUtil.getClientModel(((ClassType) rawResponseType).getName());
+                    List<ClientModelProperty> properties = new ArrayList<>();
+                    traverseProperties(model, properties);
+                    responseBodyType = properties.stream()
+                            .filter(property -> property.getName().equals(itemName))
+                            .map(clientModelProperty -> clientModelProperty.getClientType())
+                            .map(valueListType -> {
+                                // value type is List<T>, we need to get the typeArguments
+                                IType[] listTypeArgs = ((ListType) valueListType).getTypeArguments();
+                                if (listTypeArgs.length > 0) {
+                                   return listTypeArgs[0];
+                                } else {
+                                    return null;
+                                }
+                            })
+                            .findFirst().orElse(null);
```

Comment:
how would following code handle this `null`?

same as above, if not expected, fail it; if expected, handle it correctly.

Reply:
The handling logic is here:
https://github.com/Azure/autorest.java/blob/624655b5b234bd2ccc84adff2034c58574b15559/javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplateBase.java#L109

Reply:
Then it will not generate the doc for body?

Personally I would still prefer to throw, if this is not an expected situation (so either dev need to fix if this is bug, or swagger need to fix if it is not valid).